libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -2896,6 +2896,53 @@ MAV_RESULT GCS_MAVLINK::handle_command_do_set_roi(const Location &roi_loc)
     return MAV_RESULT_ACCEPTED;
 }
 
+MAV_RESULT GCS_MAVLINK::handle_command_int_do_set_home(const mavlink_command_int_t &packet)
+{
+    if (is_equal(packet.param1, 1.0f)) {
+        // if param1 is 1, use current location
+        if (!set_home_to_current_location(true)) {
+            return MAV_RESULT_FAILED;
+        }
+        return MAV_RESULT_ACCEPTED;
+    }
+    // ensure param1 is zero
+    if (!is_zero(packet.param1)) {
+        return MAV_RESULT_FAILED;
+    }
+    if ((packet.x == 0) && (packet.y == 0) && is_zero(packet.z)) {
+        // don't allow the 0,0 position
```

Comment:
This was only in Plane and not in any other vehicle. What's the reason for not allowing 0,0? It's a perfectly valid lat/lon.

Reply:
I think it was to save GCSs from themselves.

Reply:
@peterbarker I'm not sure how it saves GCSs from themselves when all vehicles except Plane currently allow (0,0).

@WickedShell That might be true, I don't know, but we should be removing that behaviour from Plane instead of putting it additionally on other vehicles.